rsconnect/api.py

```diff
@@ -796,17 +851,21 @@ def validate_posit_server(
                 raise RSConnectException("Failed to verify with {} ({}).".format(server.remote_name, exc))
 
     @cls_logged("Making bundle ...")
-    def make_bundle(self, func: Callable[..., object], *args: object, **kwargs: object):
-        path = (
-            self.get("path", **kwargs)
-            or self.get("file", **kwargs)
-            or self.get("file_name", **kwargs)
-            or self.get("directory", **kwargs)
-            or self.get("file_or_directory", **kwargs)
-        )
-        app_id = self.get("app_id", **kwargs)
-        title = self.get("title", **kwargs)
-        app_store = self.get("app_store", *args, **kwargs)
+    def make_bundle(
+        self,
+        func: Callable[..., IO[bytes]],
+        *args: object,
+        **kwargs: object,
+        # These are the actual kwargs that appear to be present in practice
```

Comment:
Let's make these named arguments and stop accepting `**kw`.

Reply:
I just realized that removing the `kwargs` here is not so straightforward, since `func` may or may not accept the specific arguments, like `multi_notebook`. I've added a `ParamSpec` so at least the `args` and `kwargs` will be checked where `make_bundle()` is called.

But looking at how the code is used, I think we can remove this very dynamic use of `func`.

This is a typical way that `make_bundle` is used right now:
```
        ce.make_bundle(
            make_html_bundle,
            path,
            entrypoint,
            extra_files,
            exclude,
        )
```

But looking at the content of this function, I think it can be restructured to be used this way, where `make_bundle()` just gets the built bundle instead of all those args:

```py
        .make_bundle(
            make_html_bundle(
                path,
                entrypoint,
                extra_files,
                exclude,
            )
        )
```

Although there also needs to be some error handling for `make_html_bundle()`.

But also, it looks like there's some more cleanup and refactoring possible for `make_bundle()`.